src/main/java/org/squiddev/plethora/integration/astralsorcery/MetaBlocks.java

```diff
@@ -0,0 +1,365 @@
+package org.squiddev.plethora.integration.astralsorcery;
+
+import dan200.computercraft.api.lua.ILuaObject;
+import hellfirepvp.astralsorcery.AstralSorcery;
+import hellfirepvp.astralsorcery.common.block.network.BlockCollectorCrystalBase;
+import hellfirepvp.astralsorcery.common.constellation.IMinorConstellation;
+import hellfirepvp.astralsorcery.common.constellation.IWeakConstellation;
+import hellfirepvp.astralsorcery.common.item.ItemColoredLens;
+import hellfirepvp.astralsorcery.common.item.block.ItemCollectorCrystal;
+import hellfirepvp.astralsorcery.common.item.crystal.CrystalProperties;
+import hellfirepvp.astralsorcery.common.lib.BlocksAS;
+import hellfirepvp.astralsorcery.common.lib.Constellations;
+import hellfirepvp.astralsorcery.common.tile.*;
+import hellfirepvp.astralsorcery.common.tile.network.TileCollectorCrystal;
+import hellfirepvp.astralsorcery.common.tile.network.TileCrystalLens;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.item.EnumDyeColor;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.common.util.Constants;
+import org.squiddev.plethora.api.Injects;
+import org.squiddev.plethora.api.meta.BaseMetaProvider;
+import org.squiddev.plethora.api.meta.BasicMetaProvider;
+import org.squiddev.plethora.api.meta.IMetaProvider;
+import org.squiddev.plethora.api.meta.ItemStackContextMetaProvider;
+import org.squiddev.plethora.api.method.ContextHelpers;
+import org.squiddev.plethora.api.method.IPartialContext;
+import org.squiddev.plethora.api.method.TypedLuaObject;
+import org.squiddev.plethora.integration.MetaWrapper;
+import org.squiddev.plethora.utils.WorldDummy;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+// Despite the name, this class also houses the providers for TileEntities, and their item forms.
+@Injects(AstralSorcery.MODID)
+public final class MetaBlocks {
+	private MetaBlocks() {
+	}
+
+	//REFINE Enable the examples if we find a way to work around Astral's `WorldNetworkHandler` logging errors
+	// due to the tiles not being registered in the starlight network.
+	// See TileNetwork.onFirstTick
+
+	public static final IMetaProvider<TileWell> META_LIGHT_WELL = new BaseMetaProvider<TileWell>(
+		"Provides the catalyst item in a Lightwell"
+	) {
+
+		@Nonnull
+		@Override
+		public Map<String, ?> getMeta(@Nonnull IPartialContext<TileWell> context) {
+			//REFINE Do we want this stack wrapped, or should we just `getMeta`?
+			ILuaObject stack = ContextHelpers.wrapStack(context, context.getTarget().getInventoryHandler().getStackInSlot(0));
+			return stack != null
+				? Collections.singletonMap("catalyst", stack)
+				: Collections.emptyMap();
+		}
+
+		@Nullable
+		@Override
+		public TileWell getExample() {
```

Comment:
Likewise, just remove for now. It's in git history should we need it.

Reply:
Fair enough, will do!